google/cloud/pubsublite/internal/containing_publisher_connection.h

```diff
@@ -0,0 +1,54 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_PUBSUBLITE_INTERNAL_CONTAINING_PUBLISHER_CONNECTION_H
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_PUBSUBLITE_INTERNAL_CONTAINING_PUBLISHER_CONNECTION_H
+
+#include "google/cloud/pubsub/publisher_connection.h"
+#include "google/cloud/options.h"
+
+namespace google {
+namespace cloud {
+namespace pubsublite_internal {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+
+// ported from google::cloud::pubsub
```

Comment:
please don't port, just move the pubsub version to a header in the pubsub internal folder, then import it